external/kafka/src/main/scala/org/apache/spark/streaming/kafka/DirectKafkaInputDStream.scala

```diff
@@ -117,6 +116,12 @@ class DirectKafkaInputDStream[
     val rdd = KafkaRDD[K, V, U, T, R](
       context.sparkContext, kafkaParams, currentOffsets, untilOffsets, messageHandler)
 
+    // Report the number of records of the batch interval to InputInfoTracker.
+    val currentNumRecords = currentOffsets.map(_._2).sum
```

Comment:
`currentNumRecords` sounds like "current number of records to be processed". That is not what this is. Also the way you are calculating this is not very intuitive. A better way would be 
```
```